src/libs/actions/Session/index.ts

```diff
@@ -50,44 +48,121 @@ import ROUTES from '@src/ROUTES';
 import SCREENS from '@src/SCREENS';
 import type Credentials from '@src/types/onyx/Credentials';
 import type {AutoAuthState} from '@src/types/onyx/Session';
+import type Session from '@src/types/onyx/Session';
 import clearCache from './clearCache';
 
-let sessionAuthTokenType: string | null = '';
-let sessionAuthToken: string | null = null;
+let session: Session = {};
 let authPromiseResolver: ((value: boolean) => void) | null = null;
-
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (session) => {
-        sessionAuthTokenType = session?.authTokenType ?? null;
-        sessionAuthToken = session?.authToken ?? null;
-
-        if (sessionAuthToken && authPromiseResolver) {
+    callback: (value) => {
+        session = value ?? {};
+        if (session.authToken && authPromiseResolver) {
             authPromiseResolver(true);
             authPromiseResolver = null;
         }
     },
 });
 
+let stashedSession: Session = {};
+Onyx.connect({
+    key: ONYXKEYS.STASHED_SESSION,
+    callback: (value) => (stashedSession = value ?? {}),
+});
+
 let credentials: Credentials = {};
 Onyx.connect({
     key: ONYXKEYS.CREDENTIALS,
     callback: (value) => (credentials = value ?? {}),
 });
 
+let stashedCredentials: Credentials = {};
+Onyx.connect({
+    key: ONYXKEYS.STASHED_CREDENTIALS,
+    callback: (value) => (stashedCredentials = value ?? {}),
+});
+
 let preferredLocale: ValueOf<typeof CONST.LOCALES> | null = null;
 Onyx.connect({
     key: ONYXKEYS.NVP_PREFERRED_LOCALE,
     callback: (val) => (preferredLocale = val),
 });
 
+function isSupportalToken(): boolean {
```

Comment:
hm, there was a reason why this was added in both places when we first implemented this. I think it was something related to cyclical imports and that the middleware couldn't use the place where we kept isSupportAuthToken before. But it's been a while since the first version, so I don't remember why anymore, and looks like I can use the Session. isSupportAuthToken everywhere.

Reply:
Nope, it actually fails to execute if I try to use `Session.isSupportAuthToken` here.